src/Place/Canonical/DBALDuplicatePlaceRepository.php

```diff
@@ -78,4 +78,57 @@ public function getDuplicatesOfPlace(string $placeId): ?array
 
         return count($duplicates) > 0 ? $duplicates : null;
     }
+
+    /** @return ClusterRecord[] */
+    public function calculateNoLongerInCluster(): array
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb->select('dp.*')
+            ->from('duplicate_places', 'dp')
+            ->leftJoin('dp', 'duplicate_places_import', 'dpi', 'dpi.cluster_id = dp.cluster_id AND dpi.place_uuid = dp.place_uuid')
+            ->where('dpi.cluster_id IS NULL')
+            ->orderBy('dp.cluster_id', 'asc')
+            ->addOrderBy('dp.place_uuid', 'asc');
+
+        $statement = $qb->execute();
+
+        return $this->processRawToClusterRecord($statement->fetchAllAssociative());
+    }
+
+    public function countPlacesInDuplicatePlacesImport(string $placeId): int
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $statement = $qb->select('count(*) as total')
+            ->from('duplicate_places_import')
+            ->where('place_uuid = :place_id')
+            ->setParameter(':place_id', $placeId)
+            ->execute();
+
+        return $statement->fetchAssociative()['total'];
+    }
+
+    public function addToDuplicatePlacesRemovedFromCluster(string $placeId): void
+    {
+        $this->connection->createQueryBuilder()
+            ->insert('duplicate_places_removed_from_cluster')
```

Comment:
I would create a new repo for this. There is also one query in `ProcessDuplicatePlaces` in the method `getDuplicatePlacesRemovedFromCluster` which can be moved inside this new repo.